src/app/components/task-form-create/task-form-create.component.ts

```diff
@@ -1,48 +1,106 @@
 import {ChangeDetectionStrategy, Component, ViewEncapsulation} from '@angular/core';
-import {FormControl, FormGroup, Validators} from '@angular/forms';
-import {Observable, switchMap, take, tap} from 'rxjs';
+import {FormArray, FormControl, FormGroup, Validators} from '@angular/forms';
+import {ActivatedRoute, Router} from '@angular/router';
+import {Observable, Subscription} from 'rxjs';
+import {switchMap, take, tap} from 'rxjs/operators';
 import {CategoryModel} from '../../models/category.model';
+import {TeamMemberModel} from '../../models/team-member.model';
 import {CategoryService} from '../../services/category.service';
 import {TaskService} from '../../services/task.service';
-import {ActivatedRoute, Router} from '@angular/router';
+import {TeamMemberService} from '../../services/team-member.service';
+import {UploadClient} from "@uploadcare/upload-client";
 
 @Component({
     selector: 'app-task-form-create',
     styleUrls: ['./task-form-create.component.scss'],
     templateUrl: './task-form-create.component.html',
     encapsulation: ViewEncapsulation.Emulated,
-    changeDetection: ChangeDetectionStrategy.OnPush
+    changeDetection: ChangeDetectionStrategy.Default
 })
 export class TaskFormCreateComponent {
     readonly form: FormGroup = new FormGroup({
         name: new FormControl('', [Validators.required]),
-        categoryId: new FormControl('', [Validators.required])
+        categoryId: new FormControl('', [Validators.required]),
+        imageUrl: new FormControl(),
+        imageFile: new FormControl(),
+        teamMembers: new FormArray([])
     });
-    private selectedId: string = ''
-    readonly categories$: Observable<CategoryModel[]> = this._activatedRoute.params
-        .pipe(switchMap((dataRoute) => {
-                    this.selectedId = dataRoute['categoryId']
-                    return this._categoryService.getAll()
-                }
-            ),
-            take(1)
-            , tap((data) => {
-                let id = data.find(x => x.id === this.selectedId)?.id
-                this.form.patchValue({categoryId: id})
+
+    get teamMembersArray() {
+        return this.form.controls['teamMembers'] as FormArray;
+    }
+    client = new UploadClient({publicKey: 'b66cb5aa61ed990132b2'})
```

Comment:
Nope, if you are using some client, which has publicKey - create service for it. Like for httpClient. Move responsibility of data logic to data layer = Service